File: 2 SEGUNDO/1 Cuatrimestre/Programación y Diseño Orientado a Objetos - PDOO/Teoria/T4/ejemplostema4java/src/CompararObjetosDefiniendoComparador/Persona.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package CompararObjetosDefiniendoComparador;

import CompararObjetosConCompareTo.*;

/**
 *
 * @author ana
 */
 
    class Persona implements Comparable<Persona>{
        String nombre;
        int edad;
        
    
    Persona(String n, int e) {
        this.nombre=n;
        this.edad=e;        
    }

    @Override
    public int compareTo(Persona per) {
       // El criterio de comparación puede ser todo lo complejo que queramos
       // en este caso va a ser la edad
       
        int resultado = 0;
        if (edad < per.edad)
            resultado = -1;
        else if(edad > per.edad)
            resultado = 1;
        return resultado;
         
    // Podéis probar cambiando el criterio de orden, por ejemplo el nombre          
    }
    
    
}

File: 2 SEGUNDO/1 Cuatrimestre/Programación y Diseño Orientado a Objetos - PDOO/Teoria/T4/ejemplostema4ruby/lib/CompararObjetosSegundaForma/persona.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
 
class Persona
  attr_reader  :edad, :nombre 
  
  def initialize(nombre,edad)
    @nombre= nombre
    @edad = edad 
  end
  def <=>per
    @edad<=>per.edad
  end
  # criterio de orden el nombre
#  def <=> per
#    @nombre <=> per.nombre
#  end
end
